src/mem/localalloc.h

```diff
@@ -252,30 +274,85 @@ namespace snmalloc
       return local_cache.remote_allocator;
     }
 
-  public:
-    constexpr LocalAllocator()
-    : handle(SharedStateHandle::get_handle()),
-      local_cache(&handle.unused_remote)
+    /**
+     * SFINAE helper.  Matched only if `T` implements `is_initialised`.  Calls
+     * it if it exists.
+     */
+    template<typename T>
+    SNMALLOC_FAST_PATH auto call_is_initialised(T*, int)
+      -> decltype(T::is_initialised())
+    {
+      return T::is_initialised();
+    }
+
+    /**
+     * SFINAE helper.  Matched only if `T` does not implement `is_initialised`.
+     * Unconditionally returns true if invoked.
+     */
+    template<typename T>
+    SNMALLOC_FAST_PATH auto call_is_initialised(T*, long)
+    {
+      return true;
+    }
+
+    /**
+     * Call `SharedStateHandle::is_initialised()` if it is implemented,
+     * unconditionally returns true otherwise.
+     */
+    SNMALLOC_FAST_PATH
+    bool is_initialised()
+    {
+      return call_is_initialised<SharedStateHandle>(nullptr, 0);
+    }
+
+    /**
+     * SFINAE helper.  Matched only if `T` implements `ensure_init`.  Calls it
+     * if it exists.
+     */
+    template<typename T>
+    SNMALLOC_FAST_PATH auto call_ensure_init(T*, int)
+      -> decltype(T::ensure_init())
+    {
+      T::ensure_init();
+    }
+
+    /**
+     * SFINAE helper.  Matched only if `T` does not implement `ensure_init`.
+     * Does nothing if called.
+     */
+    template<typename T>
+    SNMALLOC_FAST_PATH auto call_ensure_init(T*, long)
```

Comment:
Is the `T*` only required to make the template parameter used?  Does `UNUSED(T);` work if that is the case?